src/main/java/org/spongepowered/api/entity/living/player/PlayerController.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.living.player;
+
+import org.spongepowered.api.command.source.RemoteSource;
+import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.item.inventory.Container;
+import org.spongepowered.api.item.inventory.Inventory;
+import org.spongepowered.api.plugin.PluginContainer;
+import org.spongepowered.api.text.channel.ChatTypeMessageReceiver;
+import org.spongepowered.api.text.chat.ChatVisibility;
+
+import java.util.Optional;
+
+/**
+ * The controller handles certain aspects of the player.
+ *
+ * <p>The server can see and manage all players, but the client can only manage
+ * itself. e.g. The client can show itself a container, but it cannot do the
+ * same to someone else.</p>
+ *
+ * <p>Other methods which could be located here include things the player knows
+ * about itself, but cannot possibly know about another. If the server is not
+ * likely to know it, look elsewhere.</p>
+ *
+ * TODO I don't like the name 'controller'
+ */
+public interface PlayerController extends RemoteSource, ChatTypeMessageReceiver {
```

Comment:
Not sure if this should extend `RemoteSource` - the `ClientPlayerController` isn't technically "remote".